flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogAggregatingState.java

```diff
@@ -67,18 +71,20 @@ public OUT get() throws Exception {
     @Override
     public void add(IN value) throws Exception {
         delegatedState.add(value);
+        changeLogger.stateUpdated(delegatedState.getInternal(), getCurrentNamespace());
```

Comment:
That's interesting. Using `update` to simulate `add` could work, but I guess this is something different than what we want?

It is possible the case that "add" could be small, but "update" is huge.

Reply:
I guess here we have to have both IN serializer and ACC serializer.

Reply:
We don't know how the accumulator was affected by the added value, so we have to log it (using `logger.stateUpdated(delegated.getInternal())`).
And as we log the accumulator state, there is no need to log the added value.

Accumulator can be either bigger or smaller than the added value (e.g. list and counter). But we don't know it and can't log only the added value (above reason).

Reply:
I've added a clarifying comment as discussed offline.